tests/ui/async-await/generator-desc.stderr

```diff
@@ -17,12 +17,10 @@ error[E0308]: mismatched types
   --> $DIR/generator-desc.rs:12:16
    |
 LL |     fun(one(), two());
-   |     ---        ^^^^^ expected opaque type, found a different opaque type
+   |     ---        ^^^^^ expected future, found a different future
    |     |
    |     arguments to this function are incorrect
    |
-   = note: expected opaque type `impl Future<Output = ()>` (opaque type at <$DIR/generator-desc.rs:5:16>)
-              found opaque type `impl Future<Output = ()>` (opaque type at <$DIR/generator-desc.rs:6:16>)
    = help: consider `await`ing on both `Future`s
    = note: distinct uses of `impl Trait` result in different opaque types
```

Comment:
Maybe this message should also be specialized for futures.

```suggestion
   = note: each `async fn` produces a future that is a unique type
```

That message sucks but something like that.

Reply:
I agree with you, but this suggestion looks unrelated to this PR? maybe the right position is a followup PR?